panels/models/attraction.py

```diff
@@ -663,24 +663,40 @@ def start_time_label(self):
         return 'unknown start time'
 
     @property
-    def checkin_time(self):
+    def checkin_start_time(self):
         advance_checkin = self.attraction.advance_checkin
         if advance_checkin < 0:
-            return self.end_time
+            return self.start_time
         else:
             return self.start_time - timedelta(seconds=advance_checkin)
 
     @property
-    def checkin_time_label(self):
-        checkin = self.checkin_time_local
+    def checkin_end_time(self):
+        advance_checkin = self.attraction.advance_checkin
+        if advance_checkin < 0:
+            return self.end_time
+        else:
+            return self.start_time
```

Comment:
Isn't this a problem for the attractions that want to NOT let attendees check in right beforehand?

Reply:
Unless I've misunderstood (please correct me if I have) those attractions would configure things so that attendees must check in XX minutes beforehand, right?

Reply:
The notifications will give the desired checkin, but if someone is late, and the event hasn't started yet, I wanted the event to still be able to check people in. Which I guess they could anyway, but this seemed like a safe option.

Reply:
Oh wait, I think I see what you're saying - basically that under this logic, no matter what the XX minutes beforehand that's specified, all attractions would still allow people to check in 1 minute before the attraction.

Is that not okay?  I guess it's not desirable, but if we wanted to do otherwise then we'd need a much bigger change.  Because as of right now we only have a setting for each feature that says how many minutes beforehand checkin STARTS.  We would need a whole other database column that was like, "And here's how long before the event the checkin period lasts" which we don't have right now.

Unless I'm misunderstanding, which I certainly may be.

Reply:
You have it right